Firestore/core/test/unit/model/mutation_test.cc

```diff
@@ -51,110 +50,112 @@ using testutil::WrapObject;
 
-  EXPECT_EQ(result,
-            Doc("collection/key", 0, Map("foo", Map("bar", "new-bar-value")),
-                DocumentState::kLocalMutations));
+  EXPECT_EQ(doc,
+            Doc("collection/key", 0, Map("foo", Map("bar", "new-bar-value")))
+                .SetHasLocalMutations());
 }
 
 TEST(MutationTest, DeletesValuesFromTheFieldMask) {
-  Document base_doc =
+  MutableDocument doc =
       Doc("collection/key", 0,
           Map("foo", Map("bar", "bar-value", "baz", "baz-value")));
 
   Mutation patch = MergeMutation("collection/key", Map(), {Field("foo.bar")});
-  auto result = patch.ApplyToLocalView(base_doc, now);
+  patch.ApplyToLocalView(doc, now);
 
-  EXPECT_EQ(result,
-            Doc("collection/key", 0, Map("foo", Map("baz", "baz-value")),
-                DocumentState::kLocalMutations));
+  EXPECT_EQ(doc, Doc("collection/key", 0, Map("foo", Map("baz", "baz-value")))
+                     .SetHasLocalMutations());
 }
 
 TEST(MutationTest, PatchesPrimitiveValue) {
-  Document base_doc =
+  MutableDocument doc =
       Doc("collection/key", 0, Map("foo", "foo-value", "baz", "baz-value"));
 
   Mutation patch =
       PatchMutation("collection/key", Map("foo.bar", "new-bar-value"));
-  auto result = patch.ApplyToLocalView(base_doc, now);
+  patch.ApplyToLocalView(doc, now);
 
-  EXPECT_EQ(result,
+  EXPECT_EQ(doc,
             Doc("collection/key", 0,
-                Map("foo", Map("bar", "new-bar-value"), "baz", "baz-value"),
-                DocumentState::kLocalMutations));
+                Map("foo", Map("bar", "new-bar-value"), "baz", "baz-value"))
+                .SetHasLocalMutations());
 }
 
 TEST(MutationTest, PatchingDeletedDocumentsDoesNothing) {
-  NoDocument base_doc = testutil::DeletedDoc("collection/key", 0);
+  MutableDocument doc = testutil::DeletedDoc("collection/key", 0);
 
   Mutation patch = PatchMutation("collection/key", Map("foo", "bar"));
-  auto result = patch.ApplyToLocalView(base_doc, now);
+  patch.ApplyToLocalView(doc, now);
 
-  EXPECT_EQ(result, base_doc);
+  EXPECT_EQ(doc, testutil::DeletedDoc("collection/key", 0));
 }
 
 TEST(MutationTest, AppliesLocalServerTimestampTransformToDocuments) {
-  Document base_doc =
+  MutableDocument doc =
       Doc("collection/key", 0,
           Map("foo", Map("bar", "bar-value"), "baz", "baz-value"));
 
   Mutation transform = PatchMutation("collection/key", Map(),
                                      {{"foo.bar", ServerTimestampTransform()}});
-  auto result = transform.ApplyToLocalView(base_doc, now);
+  transform.ApplyToLocalView(doc, now);
 
   // Server timestamps aren't parsed, so we manually insert it.
   ObjectValue expected_data =
       WrapObject("foo", Map("bar", "<server-timestamp>"), "baz", "baz-value");
-  expected_data =
-      expected_data.Set(Field("foo.bar"), FieldValue::FromServerTimestamp(now));
+  expected_data.Set(Field("foo.bar"),
+                    EncodeServerTimestamp(now, absl::nullopt));
```

Comment:
Not sure if you have another PR to fix this, but `now` has type `Timestamp`, but `EncodeServerTimestamp` requires `google_protobuf_Timestamp`.

Reply:
Yes - https://github.com/firebase/firebase-ios-sdk/pull/7991/files#diff-8ca9b7524451f1805e18521af3ffeec7b82dc053437f2f55c4d127259eaa2e5eL32